### PR TITLE
Fix/enum in predicate

### PR DESCRIPTION
**Pull request by HuiSF**
*Issue #, if available:* #722 

*Description of changes:*

Originally enum type value was not correctly serialized while serializing a predicate for query operation. Changes for the fix:

* Add `isEnum` util function
* Check if a value is enum type and try to serialize if it's true

This PR converts 3 files from CRLF to LF mode to align with other files in the codebase. Please check the [second commit](https://github.com/aws-amplify/amplify-flutter/commit/56346eba1bd50711df715e22e3521b670edc4a55) to see the actual changes. Thanks!

----

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: packages/amplify_datastore_plugin_interface/lib/src/types/utils/parsers.dart [hunk 1]
```diff
@@ -17,10 +17,22 @@
...
+  }
+
+  return describeEnum(obj);
+}
 
 // only to be used internally by amplify-flutter library
 T? enumFromString<T>(String? key, List<T> values) =>
     values.firstWhereOrNull((v) => key == enumToString(v));
+
+/// check wehather an dynamic type [value] is an enum
```
Comment (cshfang) [inline packages/amplify_datastore_plugin_interface/lib/src/types/utils/parsers.dart:34]: Nit:
```suggestion
/// check whether a dynamic type [value] is an enum
```

**Review by cshfang — APPROVED** [review]

File: packages/amplify_datastore_plugin_interface/lib/src/types/utils/parsers.dart [hunk 2]
```diff
@@ -1,26 +1,38 @@
...
+  }
+
+  return describeEnum(obj);
+}
+
+// only to be used internally by amplify-flutter library
+T? enumFromString<T>(String? key, List<T> values) =>
+    values.firstWhereOrNull((v) => key == enumToString(v));
+
+/// check whether a dynamic type [value] is an enum
```
Comment (ragingsquirrel3) [inline packages/amplify_datastore_plugin_interface/lib/src/types/utils/parsers.dart:34]: nit/question: should this comment say "type [data]" or should the variable below be renamed "value" in line 35, or am I missing something?

**Review by ragingsquirrel3 — APPROVED** [review]
🚢  with small question about a comment/variable name.

File: packages/amplify_datastore_plugin_interface/lib/src/types/utils/parsers.dart (hunk 2 above)
Comment (HuiSF) [reply packages/amplify_datastore_plugin_interface/lib/src/types/utils/parsers.dart:34]: Oops mean to rename to `value`, will update

**Pull request merged**